docs/design-documents/platform/crash-reporting/crash_reporting.md

```diff
@@ -0,0 +1,276 @@
+
+### Requirements and assumptions
+
+This feature requires 256 bytes of dedicated RAM allocated for storing the error and fault context information.
+
+# System architecture and high-level design
+
+Below are the high-level goals for "Crash Reporting" feature:
+
+**Error information collection including exception context**
+
+The current error handling implementation in MbedOS already collects error and exception context. With this feature the above mentioned data structures should be placed in an uninitialized RAM region so that the data is retained after an auto-reboot(warm-reset).
+
+**Mechanism to auto reboot(also called warm-reset) the system without losing RAM contents where error info is stored**
+
+Implement auto-reboot functionality to reboot the system automatically after a fatal error. Note that the auto-reboot feature should be configurable.
+
+**Mechanism to retrieve the error data after reboot**
+
+Provide platform APIs to read the saved error information after reboot.
+
+**Mechanism to report the error data after reboot**
+
+During reboot the system should check if the reboot is caused by a fatal error and report the same using callback mechanism. 
+
+**Implementation should provide a mechanism to prevent constant reboot loop by limiting the number of auto-reboots**
+
+System should implement mechanism to track number of times the system has auto-rebooted and be able to stop auto-reboot when a configurable limit is reached.
+
+**Implementation should provide following configuration options**
+
+1. Configuration option to enable or disable this feature
+1. Configuration option to enable or disable auto-reboot when the system enters a fatal error scenario
+1. Configuration option to limit the number of auto-reboots
+1. Configuration option to print the saved error report on reboot
+
+### System architecture and component interaction
+
+The below diagram shows overall architecture of crash-reporting implementation.
+
+![System architecture and component interaction](./diagrams/crash-report-seq.jpg)
```

Comment:
Excetion Handler -> Exception typo (second blue rectangle from the top here)

Reply:
Exmaple -> example - yellow rectangle about WEAK attribute

Reply:
Would specify that reboot is specifically a warm-reboot.